app/javascript/src/utilities/test-utils.ts

```diff
@@ -86,13 +86,23 @@ function assertInputs (wrapper: ReactWrapper<unknown>, inputs: { id: string; pre
   return inputs.every(({ id, present }) => formGroups.exists(`[htmlFor="${id}"]`) == present)
 }
 
+/**
+ * Use when mounting a component with promises or side effects, such as BraintreeForm#onSubmit
+ */
+async function waitForPromises (): Promise<void> {
+  await act(async () => {
+    await new Promise((resolve) => setTimeout(resolve))
+  })
+}
+
```

Comment:
May you explain this? I don't understand it. AIUI this won't return a `Promise` b/c the only call in the function is preceded by an `await`. Besides, there's no `return` keyword. What does `act()` do?

Reply:
`async / await` is syntactic sugar for wrapping something with a promise. Not returning means the method returns `undefined`. But if method awaits something inside, then it has to be marked as async and it will wrap this undefined inside a Promise, therefore it will return `Promise<void>`. In this scenario, a `return` statement is redundant.

Does that make sense?

Reply:
Yes, but still don't understand the code. What does `act()` do? and why calling `setTimeout`?

Reply:
Anything that triggers a change in the component has to be wrapper around [act](https://reactjs.org/docs/test-utils.html#act).

`setTimeout` is used to add a delay to the promise so that it doesn't resolve immediately.